app/src/main/java/com/sdp13epfl2021/projmag/model/ImmutableProfile.kt

```diff
@@ -21,18 +21,19 @@ class ImmutableProfile private constructor(val lastName : String, val firstName
          * @param gender : gender on the profile
          * @param sciper : sciper on the profile
          * @param phoneNumber : phone number on the profile
+         * @param role : role on the profile
          * @return Success(profile) or failure(String explanation)
          */
         fun build(lastName : String, firstName : String,
-                  age : Int, gender : Gender, sciper : Int, phoneNumber :String) : Result<ImmutableProfile> {
+                  age : Int, gender : Gender, sciper : Int?, phoneNumber :String, role : Role) : Result<ImmutableProfile> {
             return when{
                 lastName.length > MAX_LAST_NAME_SIZE ->  Failure("last name is more than $MAX_LAST_NAME_SIZE characters")
                 firstName.length > MAX_FIRST_NAME_SIZE -> Failure("first name is more than $MAX_FIRST_NAME_SIZE characters")
                 age > MAX_AGE -> Failure("age is more than $MAX_AGE")
                 age <  MIN_AGE -> Failure("age is less than $MIN_AGE")
-                sciper < 0 -> Failure("sciper can't be a negative value")
+                (sciper!! < 0) -> Failure("sciper can't be a negative value")
```

Comment:
If we allow null sciper, this check will throw a NullPointerException for null value
```suggestion
                (sciper != null && sciper < 0) -> Failure("sciper can't be a negative value")
```